metrics/firebase/firestore/rules/firestore.rules

```diff
@@ -0,0 +1,68 @@
+rules_version = '2';
+service cloud.firestore {
+  match /databases/{database}/documents {
+    
+    match /projects/{projectId} {
+      allow read, delete: if isLoggedIn();
+      allow create, update: if isLoggedIn() && hasName();
+    }
+
+
+    match /build/{buildId} {
+      allow read, delete: if isLoggedIn();
+      allow create, update: if isLoggedIn() && buildIsValid(database);
+    }
+  }
+
+
+  /// Checks if request is autorised.
+  function isLoggedIn() {
+    return request.auth.uid != null
+  }
+
+  /// Checks whether data from request contains name string.
+  function hasName(){
+    let requestData = request.resource.data;
+
+    return "name" in requestData && requestData.name is string;
+  }
+
+  /// Checks whether build data from request is valid.
+  function buildIsValid(database){
+    let requestData = request.resource.data;
+
+    return projectExists(database) 
```

Comment:
Should we check that there is no extra data (additional fields that we don't expect?)